lib/rex/post/meterpreter/extensions/stdapi/sys/process.rb

```diff
@@ -319,6 +319,35 @@ def path
     return get_info()['path']
   end
 
+  #
+  # Read data from the process channel
+  #
+  def read_data(time_out_start = 0, time_out = -1)
```

Comment:
Would it make more sense for convention to implement the timeout functionality by adding [`#timed_read`](https://github.com/rapid7/rex-core/blob/master/lib/rex/io/stream.rb#L177) to [`Rex::Post::Meterpreter::Channels::Pools::StreamPool`](https://github.com/rapid7/metasploit-framework/blob/master/lib/rex/post/meterpreter/channels/pools/stream_pool.rb). That would provide a more consistent interface with other streams in the framework.

It looks like `#timed_read` might be raising an exception though, in which case [`#get`](https://github.com/rapid7/rex-core/blob/master/lib/rex/io/stream.rb#L232) might be more similar to what we need here.

The caller would still need to chomp and close though like they're doing now.

Reply:
I'm going to mark this as resolved and instead favor the [other](https://github.com/rapid7/metasploit-framework/pull/15111#discussion_r623121630) approach.